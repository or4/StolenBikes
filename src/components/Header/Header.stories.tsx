import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Header } from './Header';

storiesOf('Header', module).add('Header', () => <Header />);
