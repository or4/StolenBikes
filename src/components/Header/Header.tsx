import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { HeaderTextContainer } from 'components/HeaderTextContainer';
import { media } from 'ui/media';

// require - storybook cannot find module by import
const logo = require('components/assets/logo.png');

const Container = styled(Link)`
    display: flex;
    align-items: center;
	cursor: pointer;

	${media.mobile`
    	height: 60px;
  	`}

	${media.tablet`
    	height: 100px;
	`}

    ${media.desktop`
    	height: 140px;
	`}
`;

const Logo = styled.img`
    width: 64px;
    height: 64px;

    ${media.tabletNDesktop`
		padding: 0 18px;
	`}
`;

export const Header = () => (
    <Container to={'/'}>
        <Logo src={logo} />
        <HeaderTextContainer />
    </Container>
);
