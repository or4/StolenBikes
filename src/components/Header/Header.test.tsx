import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';

describe('Header', () => {
    it('should render correctly', () => {
        const component = <Header />;

        const container = shallow(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });
});
