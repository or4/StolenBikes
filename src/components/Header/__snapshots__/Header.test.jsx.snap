// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<Styled(Link)
  to="/"
>
  <styled.img
    src="logo.png"
  />
  <HeaderTextContainer />
</Styled(Link)>
`;
